### PR TITLE
add support for callook.info

### PR DESCRIPTION
**Pull request by classabbyamp**
also updated workflows to dedup

### CODE REVIEW

File: callsignlookuptools/callook/callook.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+class CallookDataModel(BaseModel):
+    status: enums.CallookStatus = enums.CallookStatus.VALID
+    type: enums.CallsignType = enums.CallsignType.NONE
+    current: CallookCallsignModel = CallookCallsignModel()
+    previous: CallookCallsignModel = CallookCallsignModel()
+    trustee: CallookTrusteeModel = CallookTrusteeModel()
+    name: Optional[str] = None
+    address: CallookAddressModel = CallookAddressModel()
+    location: CallookLocationModel = CallookLocationModel()
+    otherInfo: CallookOtherInfoModel = CallookOtherInfoModel()
```
Comment (0x5c) [inline callsignlookuptools/callook/callook.py:91]: Needs proper default definition for mutable types
https://pydantic-docs.helpmanual.io/usage/models/#field-with-dynamic-default-value

**Review by 0x5c — CHANGES_REQUESTED** [review]

File: callsignlookuptools/callook/callookasync.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        return self._process_search(
+            query=callsign.upper(),
+            resp=await self._do_query(
+                callsign=callsign.upper()
+            )
+        )
+
+    async def _do_query(self, **query) -> bytes:  # type: ignore
+        if self._session is None:
+            await self.start_session()
```
[the remark this answers is not in the document]
Comment (classabbyamp) [reply callsignlookuptools/callook/callookasync.py:40]: can't: no coroutine calling in init. will make a new() classmethod to do it instead

File: callsignlookuptools/callook/callookasync.py [hunk 3]
```diff
@@ -36,9 +47,11 @@ async def search(self, callsign: str) -> dataclasses.CallsignData:  # type: igno
...
-                raise exceptions.CallsignLookupError(f"Unable to connect to Callook (HTTP Error {resp.status})")
-            return await resp.read()
+        if self._session is not None:
+            async with self._session.get(self._base_url.format(query["callsign"])) as resp:
+                if resp.status != 200:
+                    raise exceptions.CallsignLookupError(f"Unable to connect to Callook (HTTP Error {resp.status})")
+                return await resp.read()
+        else:
+            raise exceptions.CallsignLookupError(("Session not initialised. "
+                                                  "Hint: Call `.start_session()` or use the `new()` classmethod."))
```
Comment (0x5c) [inline callsignlookuptools/callook/callookasync.py:57]: nit: "call x *once*" ?

**Review by 0x5c — COMMENTED** [review]
lgtm for now

**Review by 0x5c — APPROVED** [review]

**Pull request merged**